test/test_utils.py

```diff
@@ -268,3 +293,42 @@ def test_refused_response_no_question(self):
         r.set_rcode(dns.rcode.REFUSED)
         r.question = []
         utils.dnsmsg2log(r)
+
+
+class TestSSLContext(unittest.TestCase):
+    def setUp(self):
+        self._CA = TEST_CA
+        self._CA_serial = "E198832A55D5B708"
+
+    @unittest.skipUnless(ssl.HAS_ALPN, "Requires >=openssl-1.0.2")
+    @unittest.skipUnless(ssl.HAS_NPN, "Requires >=openssl-1.0.2")
+    def test_insecure_context(self):
+        """
+        Test that insecure flag creates a context where verify method is
+        CERT_NONE
+        """
+        sslctx = utils.create_custom_ssl_context(insecure=True)
+        self.assertEqual(sslctx.verify_mode, ssl.VerifyMode.CERT_NONE)
+
+    @unittest.skipUnless(ssl.HAS_ALPN, "Requires >=openssl-1.0.2")
+    @unittest.skipUnless(ssl.HAS_NPN, "Requires >=openssl-1.0.2")
+    def test_secure_context(self):
+        """
+        Test that if insecure is False, the ssl context created has
+        CERT_REQUIRED as the verify method
+        """
+        sslctx = utils.create_custom_ssl_context(insecure=False)
+        self.assertEqual(sslctx.verify_mode, ssl.VerifyMode.CERT_REQUIRED)
+
+    @unittest.skipUnless(ssl.HAS_ALPN, "Requires >=openssl-1.0.2")
+    @unittest.skipUnless(ssl.HAS_NPN, "Requires >=openssl-1.0.2")
+    def test_cafile(self):
+        with tempfile.NamedTemporaryFile(delete=False) as ca:
```

Comment:
you can let the temp file being deleted as it only needs to be on disk during the `with` context right?

Reply:
Ah, yes that was there only for testing. I'll amend that!